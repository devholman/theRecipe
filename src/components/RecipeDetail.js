import React from 'react';
import propTypes from 'prop-types';
//allows to pass many classnames as arguments. if classname is null, it won't render on screen.
import classNames from 'classnames';
import { Link } from 'react-router-dom';

const RecipeDetail = props => {
  if (!props.recipe) {
    return (
      <p
        style={props.style}
        className={classNames('h3 italic center p2 bg-white', props.className)}
      >
        Select recipe from list to see details.
      </p>
    );
  }
  const {
    id,
    name,
    image,
    ingredients,
    steps,
    category,
    calories
  } = props.recipe;

  return (
    <div
      style={props.style}
      className={classNames('bg-white p2', props.className)}
    >
      <h2 className="h2">{name}</h2>
      <img src={image} className="fit max-width-2 " />
      <div>
        <span>{category}</span>
        <span>{calories}</span>
      </div>
      <h3>{ingredients}</h3>
      {ingredients && (
        <ul className="list-reset">
          {ingredients.map(ingredient => {
            return <li key={ingredient}>{ingredient}</li>;
          })}
        </ul>
      )}
      <h3>Steps</h3>
      {steps && (
        <ol>
          {steps.map(step => {
            return <li key={step}>{step}</li>;
          })}
        </ol>
      )}
      {props.showLearnMore ? <Link to={`/recipe/${id}`}>See More</Link> : null}
    </div>
  );
};

RecipeDetail.propTypes = {
  style: propTypes.object,
  recipe: propTypes.object,
  className: propTypes.string,
  showLearnMore: propTypes.bool
};

RecipeDetail.defaultProps = {
  recipe: {}
};
export default RecipeDetail;
