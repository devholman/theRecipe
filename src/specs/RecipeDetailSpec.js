import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import RecipeDetail from '../components/RecipeDetail';

describe('<RecipeDetail />', () => {
  let testRecipe;

  beforeEach(() => {
    testRecipe = {
      id: 1,
      name: 'Chocolate Cake',
      category: 'dessert',
      ingredients: ['ing 1, ing 2'],
      steps: ['step 1', 'step 2']
    };
  });
  test('should render zero state', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail />
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  test('should correctly render a recipe', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} />
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  test('should correctly assign a className', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} className="classname-test" />
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  test('should render without recipe steps', () => {
    delete testRecipe.ingredients;
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} />
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  test('should render without recipe ingredients', () => {
    delete testRecipe.steps;
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} />
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
