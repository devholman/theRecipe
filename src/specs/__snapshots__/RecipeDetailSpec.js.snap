// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDetail /> should correctly assign a className 1`] = `
<div
  className="bg-white p2 classname-test"
>
  <h2
    className="h2"
  >
    Chocolate Cake
  </h2>
  <img
    className="fit max-width-2 "
  />
  <div>
    <span>
      dessert
    </span>
    <span />
  </div>
  <h3>
    ing 1, ing 2
  </h3>
  <ul
    className="list-reset"
  >
    <li>
      ing 1, ing 2
    </li>
  </ul>
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      step 1
    </li>
    <li>
      step 2
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail /> should correctly render a recipe 1`] = `
<div
  className="bg-white p2"
>
  <h2
    className="h2"
  >
    Chocolate Cake
  </h2>
  <img
    className="fit max-width-2 "
  />
  <div>
    <span>
      dessert
    </span>
    <span />
  </div>
  <h3>
    ing 1, ing 2
  </h3>
  <ul
    className="list-reset"
  >
    <li>
      ing 1, ing 2
    </li>
  </ul>
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      step 1
    </li>
    <li>
      step 2
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail /> should render without recipe ingredients 1`] = `
<div
  className="bg-white p2"
>
  <h2
    className="h2"
  >
    Chocolate Cake
  </h2>
  <img
    className="fit max-width-2 "
  />
  <div>
    <span>
      dessert
    </span>
    <span />
  </div>
  <h3>
    ing 1, ing 2
  </h3>
  <ul
    className="list-reset"
  >
    <li>
      ing 1, ing 2
    </li>
  </ul>
  <h3>
    Steps
  </h3>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail /> should render without recipe steps 1`] = `
<div
  className="bg-white p2"
>
  <h2
    className="h2"
  >
    Chocolate Cake
  </h2>
  <img
    className="fit max-width-2 "
  />
  <div>
    <span>
      dessert
    </span>
    <span />
  </div>
  <h3 />
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      step 1
    </li>
    <li>
      step 2
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail /> should render zero state 1`] = `
<div
  className="bg-white p2"
>
  <h2
    className="h2"
  />
  <img
    className="fit max-width-2 "
  />
  <div>
    <span />
    <span />
  </div>
  <h3 />
  <h3>
    Steps
  </h3>
  <a
    href="/recipe/undefined"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;
